Description: Add vt.handoff=7 to non-recovery Linux entries
 We do it this way rather than by changing GRUB_CMDLINE_LINUX_DEFAULT
 directly because this saves users from having to resolve configuration file
 conflicts, especially since this may not be the final version of the
 parameter.
Author: Colin Watson <cjwatson@ubuntu.com>
Author: Andy Whitcroft <apw@canonical.com>
Forwarded: not-needed
Last-Update: 2012-09-06

diff -ur grub-2.00/util/grub.d/10_linux.in grub-2.00.new/util/grub.d/10_linux.in
--- grub-2.00/util/grub.d/10_linux.in	2013-01-24 12:23:34.876894118 +0800
+++ grub-2.00.new/util/grub.d/10_linux.in	2013-01-24 12:36:22.760326756 +0800
@@ -78,6 +78,12 @@
 
 title_correction_code=
 
+for word in $GRUB_CMDLINE_LINUX_DEFAULT; do
+  if [ "$word" = splash ]; then
+    GRUB_CMDLINE_LINUX_DEFAULT="$GRUB_CMDLINE_LINUX_DEFAULT \$vt_handoff"
+  fi
+done
+
 linux_entry ()
 {
   os="$1"
@@ -109,13 +115,14 @@
       save_default_entry | sed -e "s/^/\t/"
   fi
 
+
   # Use ELILO's generic "efifb" when it's known to be available.
   # FIXME: We need an interface to select vesafb in case efifb can't be used.
   if [ "x$GRUB_GFXPAYLOAD_LINUX" != x ]; then
       if [ "x$GRUB_GFXPAYLOAD_LINUX" != xtext ]; then
 	  echo "	load_video" | sed "s/^/$submenu_indentation/"
       fi
-      echo "	set gfxpayload=$GRUB_GFXPAYLOAD_LINUX" | sed "s/^/$submenu_indentation/"
+      echo "	gfxmode $GRUB_GFXPAYLOAD_LINUX" | sed "s/^/$submenu_indentation/"
   fi
 
   echo "	insmod gzio" | sed "s/^/$submenu_indentation/"
@@ -174,6 +181,17 @@
 boot_device_id=
 title_correction_code=
 
+cat << 'EOF'
+function gfxmode {
+         set gfxpayload="${1}"
+         if [ "${1}" = "keep" ]; then
+                 set vt_handoff=vt.handoff=7
+         else
+                 set vt_handoff=
+         fi
+}
+EOF
+
 # Extra indentation to add to menu entries in a submenu. We're not in a submenu
 # yet, so it's empty. In a submenu it will be equal to '\t' (one tab).
 submenu_indentation=""
